tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -267,29 +267,52 @@ export class MetricsEffects implements OnInitEffects {
     ofType(actions.cardPinStateToggled),
     withLatestFrom(
       this.getVisibleCardFetchInfos(),
-      this.store.select(selectors.getEnableGlobalPins)
+      this.store.select(selectors.getEnableGlobalPins),
+      this.store.select(selectors.getShouldPersistSettings)
     ),
-    map(
-      ([
-        {cardId, canCreateNewPins, wasPinned},
-        fetchInfos,
-        enableGlobalPins,
-      ]) => {
-        if (!enableGlobalPins) {
-          return;
-        }
-        const card = fetchInfos.find((value) => value.id === cardId);
-        // Saving only scalar pinned cards.
-        if (!card || card.plugin !== PluginType.SCALARS) {
-          return;
-        }
-        if (wasPinned) {
-          this.savedPinsDataSource.removeScalarPin(card.tag);
-        } else if (canCreateNewPins) {
-          this.savedPinsDataSource.saveScalarPin(card.tag);
-        }
+    filter(
+      ([, , enableGlobalPins, shouldPersistSettings]) =>
+        enableGlobalPins && shouldPersistSettings
+    ),
+    map(([{cardId, canCreateNewPins, wasPinned}, fetchInfos]) => {
+      const card = fetchInfos.find((value) => value.id === cardId);
+      // Saving only scalar pinned cards.
+      if (!card || card.plugin !== PluginType.SCALARS) {
+        return;
+      }
+      if (wasPinned) {
+        this.savedPinsDataSource.removeScalarPin(card.tag);
+      } else if (canCreateNewPins) {
+        this.savedPinsDataSource.saveScalarPin(card.tag);
       }
-    )
+    })
+  );
+
+  private readonly loadSavedPins$ = this.actions$.pipe(
+    ofType(initAction, coreActions.pluginsListingLoaded),
+    withLatestFrom(
+      this.store.select(selectors.getEnableGlobalPins),
+      this.store.select(selectors.getShouldPersistSettings)
+    ),
+    filter(
+      ([, enableGlobalPins, shouldPersistSettings]) =>
+        enableGlobalPins && shouldPersistSettings
+    ),
+    map(() => {
```

Comment:
map() might not be the appropriate operation since you're not actually returning anything used later. Maybe tap()? It doesn't matter very much, though.

Same for addOrRemovePin$.

Reply:
+1. I think `tap` makes it much clearer that our intent is to produce side effect, not use the value. It's also what [ngrx docs](https://ngrx.io/guide/effects) uses and I think there are benefits to aligning with the docs if possible.

Reply:
Applied!